src/controller.cpp

```diff
@@ -46,6 +46,20 @@ tf2::Transform to_transform(const geometry_msgs::Pose & pose)
   return result;
 }
 
+// Returns the square distance between two points
+double distSquared(const tf2::Transform & a, const tf2::Transform & b)
+{
+  return std::pow(a.getOrigin().x() - b.getOrigin().x(), 2) +
+         std::pow(a.getOrigin().y() - b.getOrigin().y(), 2);
```

Comment:
`tf2::Vector` has a function for this called `distance2`

Reply:
I wanted to switch to this function but it seems that `distance2()` also takes the `z()` coordinate into account. Wouldn't that change the behavior of the function?

Reply:
When the path is received, the z-component is set to 0, so in practice this shouldn't be a difference: https://github.com/nobleo/path_tracking_pid/blob/816262318d90ac4d4da17a5ee013bc8bb6da80a2/src/path_tracking_pid_local_planner.cpp#L125

Reply:
Fixed!